packages/core/src/utils/indyIdentifiers.ts

```diff
@@ -0,0 +1,24 @@
+// For the definitions below see also: https://hyperledger.github.io/indy-did-method/#indy-did-method-identifiers
+export type Did = 'did'
+export type DidIndyMethod = 'indy'
+// Maybe this can be typed more strictly than string. Choosing string for now as this can be eg just `sovrin` or eg `sovrin:staging`
+export type DidIndyNamespace = string
+//  NOTE: because of the ambiguous nature - whether there is a colon or not within DidIndyNamespace this is the substring after the ***last*** colon
+export type NamespaceIdentifier = string
+
+// TODO: This template literal type can possibly be improved. This version leaves the substrings as potentially undefined
+export type IndyNamespace = `${Did}:${DidIndyMethod}:${DidIndyNamespace}:${NamespaceIdentifier}`
+
+export function createUnqualifiedIdentifier(qualifiedIdentifier: string): string {
+  const subStrings = qualifiedIdentifier.split(':')
+  return subStrings[subStrings.length - 1]
```

Comment:
I am a bit confused here so please help me understand this. 

How can we go from a qualified identifier to a credential definition id? I thought that that was the purpose of this method, along with unqualify a did and schemaId.

This method will not unqualify a credentialDefinitionId properly but maybe we are doing that somewhere else (I can't find where though).

Reply:
hmm maybe I'm getting this wrong here but I thought extracting the `6cgbu8ZPoWTnR5Rv5JcSMB` part from sth like `did:indy:sovrin:staging:6cgbu8ZPoWTnR5Rv5JcSMB` would be what we want here?

Reply:
Well yes that is what this method is doing. But is there anywehere in the code where we go from a qualified indentifier, for cred def or something, to an unqualified indentifier?

Reply:
we do that when we store the as anonCreds with qualified id and retrieve it via the LedgerApi. The api then returns the unqualified one. Updated/refactored this now anyway. Will push in a bit